tests/testthat/test-integration-fims-estimation-with-wrappers.R

```diff
@@ -0,0 +1,160 @@
+load(test_path("fixtures", "integration_test_data.RData"))
+
+test_that("deterministic test of fims", {
+  iter_id <- 1
+
+  result <- setup_and_run_FIMS_with_wrappers(
+    iter_id = iter_id,
+    om_input_list = om_input_list,
+    om_output_list = om_output_list,
+    em_input_list = em_input_list,
+    estimation_mode = FALSE
+  )
+
+  # Call report using deterministic parameter values
+  # obj$report() requires parameter list to avoid errors
+  report <- result$rep
+
+  # Compare log(R0) to true value
+  fims_logR0 <- as.numeric(result$obj$par[1])
+  expect_gt(fims_logR0, 0.0)
+  expect_equal(fims_logR0, log(om_input_list[[iter_id]]$R0))
+
+  # Compare numbers at age to true value
+  for (i in 1:length(c(t(om_output_list[[iter_id]]$N.age)))) {
+    expect_equal(report$naa[[1]][i], c(t(om_output_list[[iter_id]]$N.age))[i])
+  }
+
+  # Compare biomass to true value
+  for (i in 1:length(om_output_list[[iter_id]]$biomass.mt)) {
+    expect_equal(report$biomass[[1]][i], om_output_list[[iter_id]]$biomass.mt[i])
+  }
+
+  # Compare spawning biomass to true value
+  for (i in 1:length(om_output_list[[iter_id]]$SSB)) {
+    expect_equal(report$ssb[[1]][i], om_output_list[[iter_id]]$SSB[i])
+  }
+
+  # Compare recruitment to true value
+  fims_naa <- matrix(report$naa[[1]][1:(om_input_list[[iter_id]]$nyr * om_input_list[[iter_id]]$nages)],
```

Comment:
The output values from age 1 are identical in the test on lines 23-26. However, we need two separate tests to check the outputs from `report$naa` and `report$recruitment` since recruitment values are stored separately in the report.